### PR TITLE
[TensorRT EP] Enable more trt options

### PR DESCRIPTION
**Pull request by yf711**
This PR enables more ORT-TRTEP v2 options:

| trt_max_partition_iterations          |
| :------------------------------------ |
| trt_min_subgraph_size                 |
| trt_dla_enable                        |
| trt_dla_core                          |
| trt_engine_cache_prefix               |
| trt_dump_subgraphs                    |
| trt_force_sequential_engine_build     |
| trt_context_memory_sharing_enable     |
| trt_layer_norm_fp32_fallback          |
| trt_timing_cache_enable               |
| trt_force_timing_cache                |
| trt_detailed_build_log                |
| trt_build_heuristics_enable           |
| trt_sparsity_enable                   |
| trt_builder_optimization_level        |
| trt_auxiliary_streams                 |
| trt_tactic_sources                    |
| trt_extra_plugin_lib_paths            |
| trt_profile_min_shapes                |
| trt_profile_max_shapes                |
| trt_profile_opt_shapes                |
| trt_cuda_graph_enable                 |
| trt_cuda_graph_enable                 |
| trt_dump_ep_context_model        |
| trt_ep_context_file_path              | 
| trt_ep_context_embed_mode      |

Check [updated README](https://github.com/triton-inference-server/onnxruntime_backend/blob/8aeb6332138ff0931989688bafdb55426246f698/README.md#onnx-runtime-with-tensorrt-optimization) for more detail

### CODE REVIEW

**Review by Tabrizian — COMMENTED** [review]
Thanks for contribution!

Is it possible to add some testing for it in this PR: https://github.com/triton-inference-server/server/blob/main/qa/L0_onnx_optimization/test.sh

**Comment by yf711** [comment]
> Thanks for contribution!
> 
> Is it possible to add some testing for it in this PR: https://github.com/triton-inference-server/server/blob/main/qa/L0_onnx_optimization/test.sh

Could you grant me access to [server repo](https://github.com/triton-inference-server/server/blob/main)? I will init pull request after that

File: README.md [hunk 1]
```diff
@@ -93,6 +93,48 @@ TensorRT can be used in conjunction with an ONNX model to further optimize the p
 * `trt_engine_cache_enable`: Enable engine caching.
 * `trt_engine_cache_path`: Specify engine cache path.
 
+To explore the usage of more parameters, follow the mapping table below and check [ONNX Runtime doc](https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#execution-provider-options) for detail.
+
+> Please link to the latest ONNX Runtime binaries in CMake or build from [main branch of ONNX Runtime](https://github.com/microsoft/onnxruntime/tree/main) to enable latest options.
+
+### Parameter mapping between ONNX Runtime and Triton-ort
+
+| Parameter: key in Triton model config | Parameter: value in Triton model config                      | Corresponding TensorRT EP option in ONNX Runtime | Type   |
```
[the remark this answers is not in the document]
Comment (yf711) [reply README.md]: comment applied

File: README.md [hunk 2]
```diff
@@ -93,6 +93,48 @@ TensorRT can be used in conjunction with an ONNX model to further optimize the p
 * `trt_engine_cache_enable`: Enable engine caching.
 * `trt_engine_cache_path`: Specify engine cache path.
 
+To explore the usage of more parameters, follow the mapping table below and check [ONNX Runtime doc](https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#execution-provider-options) for detail.
+
+> Please link to the latest ONNX Runtime binaries in CMake or build from [main branch of ONNX Runtime](https://github.com/microsoft/onnxruntime/tree/main) to enable latest options.
+
+### Parameter mapping between ONNX Runtime and Triton-ort
```
[the remark this answers is not in the document]
Comment (yf711) [reply README.md]: comment applied

**Comment by dyastremsky** [comment]
> > Thanks for contribution!
> > Is it possible to add some testing for it in this PR: https://github.com/triton-inference-server/server/blob/main/qa/L0_onnx_optimization/test.sh
> 
> Could you grant me access to [server repo](https://github.com/triton-inference-server/server/blob/main)? I will init pull request after that

Thanks for being willing to add testing for this ticket! What sort of access do you need? There are pull requests from different users without permissions being granted. You should be able to fork the repository then submit the pull request.

**Comment by dzier** [comment]
@yf711 I have granted you access to submit your PR to the server repo.

**Comment by jywu-msft** [comment]
any updates? can we merge this PR first and then add tests in a separate PR?

**Comment by yf711** [comment]
Hi @dzier @Tabrizian Just a quick reminder that this PR has been finished. Could you test this PR on your CI?

**Review by dyastremsky — APPROVED** [review]

**Pull request merged**